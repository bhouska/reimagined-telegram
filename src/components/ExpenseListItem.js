import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <div>
    <p><Link to={`/edit/${id}`}>
            {description}
            </Link>
        </p>
        <p>{amount}</p>
        <p>{createdAt}</p>
        </div>
);

export default ExpenseListItem;