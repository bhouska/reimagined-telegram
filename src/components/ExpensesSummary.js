import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        Viewing {props.expenseCount} {props.expenseCount === 1 ? 'expense' : 'expenses'} totalling {numeral(props.expenseTotal/100).format('$0,0.00')}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expenseTotal: selectExpensesTotal(state.expenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);