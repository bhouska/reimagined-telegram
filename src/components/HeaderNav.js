import React from 'react';
import { NavLink } from 'react-router-dom';

const HeaderNav = () => (
    <nav>
        <ul>
            <li><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></li>
            <li><NavLink to="/create" activeClassName="is-active" exact={true}>Create Expense</NavLink></li>
            <li><NavLink to="/help" activeClassName="is-active" exact={true}>Help</NavLink></li>
        </ul>
    </nav>
);

export default HeaderNav;