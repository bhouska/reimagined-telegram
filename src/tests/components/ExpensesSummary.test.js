import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import selectExpensesTotal from '../../selectors/expenses-total';

test('should render correctly with 1 expense', () => {
    const expenseTotal = selectExpensesTotal([expenses[0]]);
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} expenseCount={1} expenseTotal={expenseTotal} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render correctly with multiple expenses', () => {
    const expenseTotal = selectExpensesTotal(expenses);
    const wrapper = shallow(<ExpensesSummary expenses={expenses} expenseCount={expenses.length} expenseTotal={expenseTotal} />);
    expect(wrapper).toMatchSnapshot();
});