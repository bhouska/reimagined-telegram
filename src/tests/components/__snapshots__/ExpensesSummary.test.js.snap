// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with 1 expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $1.95
</div>
`;

exports[`should render correctly with multiple expenses 1`] = `
<div>
  Viewing 
  3
   
  expenses
   totalling 
  $1,141.95
</div>
`;
